File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int voting =0;
        int element =0;
        for(int i =0;i<nums.length ;i++){
            if(voting==0){
                element = nums[i];
                voting++;
            }
            else if(element == nums[i]){
                voting ++;
            }
            else{
                voting --;
            }
        }
        int count_element=0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]==element){
                count_element++;

            }
        }
        if(count_element>(nums.length)/2){
            return element;
        }

        return 1;

    }
}

      

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n = nums.size();
        int element = 0;
        int count_voting = 0;

        for (int i = 0; i < n; i++) {
            if (count_voting == 0) {
                element = nums[i];
                count_voting++;
            } else if (element == nums[i]) {
                count_voting++;
            } else {
                count_voting--;
            }
        }
        int count_major_element = 0;
        for (int i = 0; i < n; i++) {
            if (nums[i] == element) {
                count_major_element++;
            }
        }
        // cheach whether this element is reallt applicable or not
        if (count_major_element > n / 2) {
            return element;
        }
        return -1;
    }
};